walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,70 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // throw new UnsupportedOperationException("TODO: Implement this method.");
+    final long MINUTES_IN_A_DAY = 1440;
+    final int START_OF_DAY = TimeRange.START_OF_DAY;
+    final int END_OF_DAY = TimeRange.END_OF_DAY;
+    int endOfTimeRange = 0;
+    TimeRange previousTimeRange = TimeRange.fromStartEnd(0,0,true);
+    Collection<TimeRange> unavailableTimeRange = new ArrayList<>();
+    Collection<TimeRange> unavailableNoOverlap = new ArrayList<>();
+    Collection<TimeRange> returnCollection = new ArrayList<>();
+
+    long durationOfRequest = request.getDuration();
+    Collection<String> attendeesOfRequest = request.getAttendees();
+
+    if(attendeesOfRequest.isEmpty()){
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }else if(durationOfRequest > MINUTES_IN_A_DAY){
+      return Arrays.asList();
+    }else{
+      for(String attendee : attendeesOfRequest){
+        for(Event event : events){
+          if(event.getAttendees().contains(attendee)){
+            unavailableTimeRange.add(event.getWhen());
+          }
+        }
+      }
+      for(TimeRange timerange : unavailableTimeRange){
+        if(previousTimeRange.contains(timerange)){
```

Comment:
Looks like this `if` statement isn't being used, so you can remove it

Reply:
I included this if condition as i needed to bypass the time range if its within another time range however theres no action that is needed to be taken. So i left the if condition blank.

Reply:
In general it's not a good practice to have blank conditions, it just makes things more confusing. In this case you can put `continue;` inside the condition instead, which will skip all the code after that and go to the next iteration. This also allows you to have less if/else branches, i.e.

```
if (previousTimeRange.contains(timerange)) {
  continue;
}
if (previousTimeRange.overlaps(timerange)) {
   ...
   continue;
}
unavailableNoOverlap.add(timerange);
previousTimeRange = timerange;
```